### ISSUE TITLE
hh:mm of rc.dateformat sets time to tomorrow's hh:mm instead of todays

### ISSUE THREAD

**Issue by reportaman** [GothenburgBitFactory/taskwarrior#2517]
I created a task, and then modified the start time using the following command:

```
task <num> mod start:11:45 rc.dateformat:hh:mm
```

According to the [table of supported ISO-8601 formats](https://taskwarrior.org/docs/dates.html), hh:mm should imply "today"'s hh:mm. For some reason, taskwarrior sets time to tomorrow's hh:mm. Check that using ``task <num> info``.

I tried it recently, and here's the info line:

```
Mon 28 Jun 2021 12:19:38 Start changed from 'Mon 28 Jun 2021 12:18:49' to 'Tue 29 Jun 2021 11:45:00'.
```

Why Tue 29 Jul?

**Comment by tbabej** [comment]
I think this might be a bug (or maybe a consequence of dates in TW being forward looking). Perhaps we should make that attribute dependant?

In any case, you can using `task 1 mod start:today+11.75h` as a workaround, which works as expected.

**Comment by reportaman** [comment]
@tbabej I actually tried that, but preferred doing the following:

```
task <num> mod start:2020-06-28T11:45 rc.dateformat:YYYY-MM-DDThh:mm
```

My zsh completion plugin will remember that, so hopefully I won't have to type the whole thing each time.

The thing with ``task <num> mod start:today+11.75h`` is that at times the ratio of hour (``0.75`` in this case is easy) can be some exercise. 

Doesn't look like its an artifact of TW being forward looking, because the page I point to clearly says "Today" as can be seen from the screenshot below.

> Btw, is there a part of document that talks in depth about forward looking behavior?

![Screen Shot 2021-06-28 at 1 14 35 PM](https://user-images.githubusercontent.com/71736629/123698433-02547600-d813-11eb-8dda-5d2d72805287.png)

**Comment by reportaman** [comment]
@tbabej Yep I get what you mean by attribute dependent. Yeah that does sound like a good idea. If the user types something is "due", then they usually mean in future. I would use it that way, not sure if everyone would like that so perhaps the meaning could be controllable by a taskrc setting like ``dateformat.engine=smart|literal``?

**Comment by tbabej** [comment]
You're reading my mind here!
